crates/slot-arithmetic/src/lib.rs

```diff
@@ -12,10 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use derive_more::{Add, From, Sub};
 use minicbor::{Decode, Decoder, Encode};
 use serde::{Deserialize, Serialize};
 
-type Slot = u64;
+#[derive(Clone, Debug, Copy, PartialEq, From, Add, Sub, PartialOrd, Eq, Serialize, Deserialize)]
```

Comment:
I’d be skeptical regarding `Add` and `Sub` since these can overflow in release builds, and that doesn’t make sense to me for slot numbers. Do we really need these traits here?

Reply:
_really_ probably not, it makes some expressions in the `EraHistory` impl leaner. 
I will see how it looks if I remove them and replace them